Yoast/YoastSEO.js
# Add filter for passive participles when preceded by certain words

## Summary

This PR can be summarized in the following changelog entry: Adds a filter to avoid incorrectly marking participles as passive when preceded by certain words. E.g., 'accomplished' in 'he was an accomplished pianist' will now correctly be marked as non-passive because of the preceding 'an'.

## Relevant technical choices

- Replaced the specific rules `hasHavingException`, `hasLeftException`, and `hasFitException` with a general rule
- Replaced `having.js` and `determiners.js` by importing word lists from `functionWords.js`
- Added positive lookahead to regex in `addWordBoundary.js` for use in `createRegexFromArray.js`

## Test instructions

This PR can be tested by following these steps:

- Use the browserified example. Don't forget to run a `grunt build:js`.
- Paste the sentence `he was an accomplished pianist`.
- Make sure it is not marked as passive.

Fixes #641
